File: myjournserver/src/main/java/com/wizardom/myjournserver/controller/response/JsonResponse.java
package com.wizardom.myjournserver.controller.response;

import lombok.Data;
import lombok.experimental.Accessors;
import org.springframework.http.HttpStatus;

import java.util.Map;

@Data
@Accessors(chain = true)
public class JsonResponse<T> {
    private T body;
    private HttpStatus status;
    private Map<String, String> errors;

    public JsonResponse(HttpStatus status){
        this.status = status;
    }

    public JsonResponse(HttpStatus status, T body){
        this.status = status;
        this.body = body;
    }

    public JsonResponse(HttpStatus status, Map<String, String> errors){
        this.status = status;
        this.errors = errors;
    }
}

File: app/lib/services/repository/jsonresponse.dart
import 'package:dio/dio.dart';

class JsonResponse<T> {
  final int statusCode;
  final Headers headers;
  final T data;
  final Map<String, dynamic> errors;

  JsonResponse(this.statusCode, this.data, this.errors, this.headers);

  factory JsonResponse.fromResponse(Response response) {
    JsonResponse jsonResponse;
    try {
      jsonResponse = JsonResponse(response.data["status"], response.data["body"], response.data["errors"], response.headers);
    return jsonResponse; }
    catch (e) {
      return JsonResponse(response.statusCode, null, null, response.headers);
    }

  }

}